Core.ApplicationServices/Organizations/OrganizationRightsService.cs

```diff
@@ -75,6 +85,64 @@ public class OrganizationRightsService : IOrganizationRightsService
             return RemoveRight(right);
         }
 
+        public Maybe<OperationError> RemoveUnitRole(int rightId)
+        {
+            return RemoveUnitRightsByIds(rightId.WrapAsEnumerable());
+        }
+
+        public Maybe<OperationError> RemoveUnitRightsByIds(IEnumerable<int> rightIds)
+        {
+            using var transaction = _transactionManager.Begin();
+            var rightsToDelete = new List<OrganizationUnitRight>();
+            foreach (var rightId in rightIds)
+            {
+                var unitRight = _unitRightRepository.GetByKey(rightId);
+                if (unitRight == null)
+                    return new OperationError($"Unit right with id: {rightId} not found", OperationFailure.NotFound);
+                if (!_authorizationContext.AllowDelete(unitRight))
+                    return new OperationError("User is not allowed to perform this operation", OperationFailure.Forbidden);
+                
+                rightsToDelete.Add(unitRight);
+            }
+            var userIds = rightsToDelete.Select(x => x.UserId).ToList();
+
+            _unitRightRepository.RemoveRange(rightsToDelete);
+            foreach (var userId in userIds.Distinct())
+            {
+                _domainEvents.Raise(new AdministrativeAccessRightsChanged(userId));
```

Comment:
I'm not sure if that's how I should raise the domain events, but I couldn't find another way when removing a range

Reply:
this is the right way